src/os/README.md

```diff
@@ -104,3 +106,65 @@ inspiration for several decisions in this library.
 - that said, if an operation is extremely efficient and almost always wanted,
   consider doing it natively, automatically. For example, ints in HSL
   `sockaddr` are always in host byte order, not network byte order.
+
+## Appendix:  C / Hack type mapping cheat sheet
+
+| C type | Hack type |
+|---|---|
+| `int` or `short` as a bit set | `keyset<Flag>` where `Flag` is an `enum` |
+| `int socket` or `int filedes`, or other long-lived system resources | `HH\Lib\OS\FileDescriptor` or other wrapper classes |
+| Setters for `void *` or `struct *` whose reference would not be held by C libraries | `vec`, see [Appendix: encoding lightweight C data types with setters](#appendix-encoding-lightweight-c-data-types-with-setters) |
+
+## Appendix: encoding lightweight C data types with setters
+
+A common practice in C library is to expose a set of functions, including a
+constructor, a destructor and several setters, to manipulate a C
+data type. If the C data type represents a native resource, e.g. `FILE *`, we
```

Comment:
Probably better to use '`int` as a file descriptor' - we don't use `FILE*`. It gives some extra features (especially buffering) over raw FD access that interfere with what HSL IO offers with O_NONBLOCK.

Reply:
Updated